assets/scripts/submit.js

```diff
@@ -10,23 +10,36 @@
   const passphraseInput = document.getElementById('password')
   const redirectInput = document.getElementById('redirect')
   const submitButton = document.getElementById('login-submit')
+  const twoFactorPasscodeInput = document.getElementById('two-factor-passcode')
+  const twoFactorTokenInput = document.getElementById('two-factor-token')
+  const twoFactorForm = document.getElementById("two-factor-form")
+  const passwordForm = document.getElementById("password-form")
 
   let errorPanel = loginForm && loginForm.querySelector('.errors')
 
   const showError = function (error) {
-    error = error || 'The Cozy server is unavailable. Do you have network?'
+    if (error) {
+      error = '' + error
+    } else {
+      error = 'The Cozy server is unavailable. Do you have network?'
+    }
 
     if (!errorPanel) {
       errorPanel = document.createElement('div')
       errorPanel.classList.add('errors')
+      errorPanel.appendChild(document.createElement('p'))
       loginForm.appendChild(errorPanel)
     }
 
-    errorPanel.innerHTML = '<p>' + error + '</p>'
```

Comment:
XSS warning here: even though the current possible errors are all independent from user input, this may not always be the case.